xrpl/model/transactions/trust_set_test.go

```diff
@@ -51,3 +51,42 @@ func TestTrustSetTx(t *testing.T) {
 		t.Error("UnmarshalTx result differs from expected")
 	}
 }
+
+func TestTrustSetFlatten(t *testing.T) {
+	s := TrustSet{
+		BaseTx: BaseTx{
+			Account:            "ra5nK24KXen9AHvsdFTKHSANinZseWnPcX",
+			TransactionType:    TrustSetTx,
+			Fee:                types.XRPCurrencyAmount(12),
+			Flags:              262144,
+			Sequence:           12,
+			LastLedgerSequence: 8007750,
+		},
+		LimitAmount: types.IssuedCurrencyAmount{
+			Issuer:   "rsP3mgGb2tcYUrxiLFiHJiQXhsziegtwBc",
+			Currency: "USD",
+			Value:    "100",
+		},
+	}
+
+	flattened := s.Flatten()
+	
+	expected := map[string]interface{}{
+		"Account":            "ra5nK24KXen9AHvsdFTKHSANinZseWnPcX",
+		"TransactionType":    "TrustSet",
+		"Fee":                uint64(12),
```

Comment:
Shouldn't Fee be a string according to the docs ?
https://xrpl.org/docs/references/protocol/transactions/common-fields#transaction-common-fields

Reply:
Nice catch!

I checked different transaction models types and there are some mismatches with the transaction docs you mentioned. 

I have updated the desired ones (`Payment`, `Clawback`, `TrustSet` and `TicketCreate`)